File: 20220913/ompp-0.8.5/opari/tool/Test/testG.f
       program testG
         integer me, glob
         integer omp_get_thread_num

         !$omp parallel private(me,glob)
           me = omp_get_thread_num()
           write(*,*) "hello from", me

           !$omp single
             glob = me
             write(*,*) "single1 from", me
           !$omp end single copyprivate(glob)

           write(*,*) "hello from", me

           !$omp single 
             write(*,*) "single2 from", me
           !$omp end single 

           !$omp single
             write(*,*) "single3 from", me
           !$omp end single nowait

         !$omp end parallel
       end program testG

File: 20220913/ompp-0.8.5/opari/tool/Test/testG.c
#include <omp.h>
#include <stdio.h>

int main() {
  int me, glob;

  #pragma omp parallel private(me,glob)
  {
    me = omp_get_thread_num();
    printf("hello from %d\n", me);

    #pragma omp single copyprivate(glob)
    {
      glob = me;
      printf("single1 from %d\n", me);
    }

    printf("hello from %d\n", me);

    #pragma omp single 
    {
      printf("single2 from %d\n", me);
    }

    #pragma omp single nowait
    {
      printf("single3 from %d\n", me);
    }

  }
}
